Erreur à l'installation

Salut ! 

En tentant d'installer ce projet sur ma machine, étant totalement débutant en Rust, j'obtiens ces erreurs. 

`error[E0425]: cannot find value `from_os_str` in this scope`
`error[E0599]: no method named `parse` found for struct `Arg` in the current scope`
![image](https://github.com/rust-transit/gtfs-to-geojson/assets/1177762/600e21ad-13f5-4262-8ad0-04525f67cb46)

Peut-être est-ce un problème de version de Rust ? Je n'ai rien trouvé sur le Web à ce propos. 

Contexte : j'utilise node-gtfs, ça marche très bien... sauf pour générer les geojson des shapes, c'est très lent. D'où ma recherche d'alternatives plus efficaces :)

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -8,11 +8,8 @@ jobs:
     name: Publish on crates.io
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-            toolchain: stable
-            override: true
-      - uses: katyo/publish-crates@v1
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: katyo/publish-crates@v2
         with:
             registry-token: ${{ secrets.CARGO_REGISTRY_TOKEN }}

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -7,8 +7,8 @@ jobs:
     name: Test Suite
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v3
         with:
           path: |
             ~/.cargo/bin/
@@ -17,36 +17,21 @@ jobs:
             ~/.cargo/git/db/
             target/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --all-features
+      - uses: katyo/publish-crates@v2
+      - run: cargo test --all-features
 
   lints:
     name: Lints
     runs-on: ubuntu-latest
     steps:
       - name: Checkout sources
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Install stable toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          components: rustfmt, clippy
+      - uses: katyo/publish-crates@v2
 
       - name: Run cargo fmt
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+        run: cargo fmt --all -- --check
 
       - name: Run cargo clippy
-        uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: -- -D warnings
+        run: cargo clippy -- -D warnings

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "gtfs-geojson"
-version = "0.4.0"
+version = "0.5.0"
 authors = ["N07070 <social@n07070.xyz>"]
 edition = "2018"
 description = "A simple converter for GTFS to GeoJSON formats"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,8 +14,8 @@ categories = ["data-structures"]
 
 [dependencies]
 
-gtfs-structures = "0.38.0"
+gtfs-structures = "0.41.0"
 geojson = "0.24.1"
-clap = { version = "4.4.11", features = ["derive"] }
+clap = { version = "4.5.1", features = ["derive"] }
 geo-types = "0.7.12"
 serde_json = "1"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,7 +2,7 @@
 name = "gtfs-geojson"
 version = "0.5.0"
 authors = ["N07070 <social@n07070.xyz>"]
-edition = "2018"
+edition = "2021"
 description = "A simple converter for GTFS to GeoJSON formats"
 readme = "README.md"
 repository = "https://gitlab.com/CodeursEnLiberte/gtfs-to-geojson/"

--- a/src/converter.rs
+++ b/src/converter.rs
@@ -123,8 +123,8 @@ pub fn get_route_properties(
 /// This function will take a GTFS data format and ouput a FeatureCollection, which can in turn, be printed by the utility module.
 /// # Examples
 /// ```
-/// let gtfs_data = Gtfs::new("tests/gtfs/gtfs_46.zip");
-/// convert_to_geojson(gtfs_data, true);
+/// let gtfs_data = gtfs_structures::Gtfs::new("test/basic/gtfs").unwrap();
+/// gtfs_geojson::convert_to_geojson(&gtfs_data);
 /// ```
 pub fn convert_to_geojson(gtfs_data: &Gtfs) -> FeatureCollection {
     let mut features = extract_stops(gtfs_data);

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,4 +1,3 @@
 mod converter;
 
 pub use converter::*;
-

--- a/src/main.rs
+++ b/src/main.rs
@@ -70,7 +70,7 @@ mod test {
             json!(given_feature.as_ref().unwrap().properties),
             json!({
             "code": "0001",
-            "description": "",
+            "description": null,
             "id": "stop2",
             "name": "StopPoint",
             "wheelchair_boarding": "unknown"
@@ -107,7 +107,7 @@ mod test {
         assert_eq!(
             json!(given_feature.as_ref().unwrap().properties),
             json!({
-                "description": "",
+                "description": null,
                 "id": "stop1",
                 "name": "Stop Area",
                 "wheelchair_boarding": "unknown"

--- a/src/main.rs
+++ b/src/main.rs
@@ -15,16 +15,14 @@ struct Opt {
         name = "gtfs",
         short = 'i',
         long = "input",
-        help = "Path to the GTFS file (can be a directory or a zip file) or URL to an online GTFS file",
-        parse(from_os_str)
+        help = "Path to the GTFS file (can be a directory or a zip file) or URL to an online GTFS file"
     )]
     file: PathBuf,
     #[clap(
         name = "output",
         short = 'o',
         long = "output",
-        help = "Path to the output file. If not present, geojson file is outputed in stdout",
-        parse(from_os_str)
+        help = "Path to the output file. If not present, geojson file is outputed in stdout"
     )]
     output_file: Option<PathBuf>,
 }

